sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -434,24 +438,31 @@ object ScalaNativePluginInternal {
       (nativeExternalDependencies.value.toSet --
         nativeAvailableDependencies.value.toSet).toList.sorted
     },
-    nativeAvailableDependencies := ResourceScope { implicit scope =>
-      val forceCompile = compile.value
-
-      val globals = fullClasspath.value
-        .collect { case p if p.data.exists => p.data }
-        .flatMap(p => tools.LinkerPath(VirtualDirectory.real(p)).globals.toSeq)
-
-      globals.map(_.show).sorted
-    },
-    nativeExternalDependencies := ResourceScope { implicit scope =>
+    nativeAvailableDependencies := Def
+      .task {
+        val fcp = fullClasspath.value
+        ResourceScope { implicit scope =>
+          val globals = fcp
+            .collect { case p if p.data.exists => p.data }
+            .flatMap(p =>
+              tools.LinkerPath(VirtualDirectory.real(p)).globals.toSeq)
+
+          globals.map(_.show).sorted
+        }
+      }
+      .dependsOn(compile)
```

Comment:
`fullClasspath` indirectly depends on `compile` for the same configuration, so this line is unnecessary. Consequently, you don't need the `Def.task` wrapper.